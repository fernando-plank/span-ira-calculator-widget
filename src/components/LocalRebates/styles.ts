import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  text-align: left;
  color: ${({ theme }) => theme.colors.black};
  background: ${({ theme }) => theme.colors.gray};
  width: 100%;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    padding: 56px 24px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileXL}) {
    padding: 56px 32px;
  }

  @media (min-width: ${(props) => props.theme.breakpoints.tablet}) {
    padding: 90px 40px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    padding: 96px 64px;
  }
`

export const Header = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: baseline;
`

export const Title = styled.h1`
  font-style: normal;
  font-weight: 700;
  font-size: 48px;
  line-height: 56px;
  margin-bottom: 81px;
`

export const Body = styled.div`
  display: flex;
  height: 100%;
  opacity: 0;
  max-height: 0;
  pointer-events: none;
  transition: max-height 0.2s ease-out;

  &.active {
    max-height: 400px;
    opacity: 1;
    pointer-events: auto;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    flex-direction: column;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    flex-direction: row;
  }
`

export const BodyWrapper = styled.div`
  display: flex;
  flex-direction: column;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    width: 100%;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    width: 50%;
  }
`

export const BodyContent = styled.div`
  display: flex;
  margin-top: 40px;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    flex-direction: column;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    flex-direction: row;
  }
`

export const BodyText = styled.p`
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  width: 100%;
`

export const LearnMoreButton = styled.button`
  border: 2px solid ${({ theme }) => theme.colors.black};
  border-radius: 20px;
  width: 50%;
  height: 40px;
  margin-left: 50px;
  font-style: normal;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    align-self: end;
    margin-top: 20px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    align-self: start;
    margin-top: 0px;
  }
`

export const LocalName = styled.h2`
  font-style: normal;
  font-weight: 700;
  font-size: 32px;
  line-height: 40px;
  margin-top: 40px;
`

export const Separator = styled.div`
  height: 1px;
  background: #b0b0b1;
  border-radius: 2px;
`

export const TextFooter = styled.p`
  font-style: normal;
  font-weight: 500;
  margin-top: 50px;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    width: 100%;
    font-size: 16px;
    line-height: 24px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    width: 50%;
    font-size: 12px;
    line-height: 16px;
  }
`

export const PlusIcon = styled.img`
  cursor: pointer;
`
