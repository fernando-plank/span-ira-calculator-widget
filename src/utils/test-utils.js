var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
/* eslint-disable import/export */
import React from 'react';
import theme from '@styles/theme';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
jest.mock('../components/Icon', () => {
    return jest.fn(() => React.createElement("div", null, "mocked icon"));
});
const customRender = (ui, _a = {}) => {
    var renderOptions = __rest(_a, []);
    return render(React.createElement(ThemeProvider, { theme: theme }, ui), renderOptions);
};
export * from '@testing-library/react';
export { customRender as render };
