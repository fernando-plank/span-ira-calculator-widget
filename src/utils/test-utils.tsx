/* eslint-disable import/export */
import React, { ReactElement } from 'react'

import theme from '@styles/theme'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

type CustomRenderProps = Omit<RenderOptions, 'queries'>

jest.mock('../components/Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

const customRender = (
  ui: ReactElement,
  { ...renderOptions }: CustomRenderProps = {}
) => render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions)

export * from '@testing-library/react'
export { customRender as render }
