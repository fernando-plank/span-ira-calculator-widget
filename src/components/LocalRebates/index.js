import React from 'react';
import Accordion from '@components/Accordion';
import * as S from './styles';
const LocalRebates = () => {
    return (React.createElement(S.Wrapper, null,
        React.createElement("div", null,
            React.createElement(S.Title, { "data-testid": "local-rebates-title" }, "Local utility electrification rebates")),
        React.createElement("div", null,
            React.createElement(Accordion, { icon: 'Plus', title: 'California' },
                React.createElement(S.BodyWrapper, null),
                React.createElement(S.BodyWrapper, null,
                    React.createElement(S.BodyContent, null,
                        React.createElement(S.BodyText, null,
                            React.createElement("b", null, "Silicon Valley Power"),
                            " customers can save up to $2,500 on SPAN as part of a larger home electrification project."),
                        React.createElement(S.LearnMoreButton, null, "Learn more")),
                    React.createElement(S.BodyContent, null,
                        React.createElement(S.BodyText, null,
                            React.createElement("b", null, "Sacramento Municipal Utility District"),
                            " (SMUD) customers can save up to $2,500 as part of a larger home electrification project."),
                        React.createElement(S.LearnMoreButton, null, "Learn more")))),
            React.createElement(Accordion, { icon: 'Plus', title: 'Hawaiâ€™i' },
                React.createElement(S.BodyWrapper, null),
                React.createElement(S.BodyWrapper, null,
                    React.createElement(S.BodyContent, null,
                        React.createElement(S.BodyText, null,
                            React.createElement("b", null, "Shifted Energy"),
                            " customers can have the costs of SPAN Panel covered, plus up to $4,000 for install when they agree to share SPAN Panel data and control with Shifted Energy."),
                        React.createElement(S.LearnMoreButton, null, "Learn more")),
                    React.createElement(S.BodyContent, null,
                        React.createElement(S.BodyText, null,
                            React.createElement("b", null, "Hawaiian Electric"),
                            " customers can save $500 on SPAN Panel product costs by becoming a HECO Utility rate payer."),
                        React.createElement(S.LearnMoreButton, null, "Learn more"))))),
        React.createElement("div", null,
            React.createElement(S.TextFooter, null, "We\u2019re constantly updating this list, so check back to see if your utility provider provides localized incentives for getting SPAN."))));
};
export default LocalRebates;
