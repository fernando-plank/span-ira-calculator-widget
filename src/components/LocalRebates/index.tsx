import React from 'react'
import Accordion from '@components/Accordion'

import * as S from './styles'

const LocalRebates = () => {
  return (
    <S.Wrapper>
      <S.Content>
        <div>
          <S.Title data-testid="local-rebates-title">
            Local utility electrification rebates
          </S.Title>
          <S.Subtitle data-testid="local-rebates-title">
            Enjoy even more potential savings from
          </S.Subtitle>
          <S.Subtitle data-testid="local-rebates-title">
            localized utility incentives.
          </S.Subtitle>
        </div>
        <div style={{ marginTop: '40px' }}>
          <Accordion icon={'Plus'} title={'California'}>
            <S.BodyWrapper></S.BodyWrapper>
            <S.BodyWrapper>
              <S.BodyContent>
                <S.BodyText>
                  <b>Silicon Valley Power</b> customers can save up to $2,500 on
                  SPAN as part of a larger home electrification project.
                </S.BodyText>
                <S.LearnMoreButton>Learn more</S.LearnMoreButton>
              </S.BodyContent>
              <S.BodyContent>
                <S.BodyText>
                  <b>Sacramento Municipal Utility District</b> (SMUD) customers
                  can save up to $2,500 on SPAN Panel as part of a larger home
                  electrification project.
                </S.BodyText>
                <S.LearnMoreButton>Learn more</S.LearnMoreButton>
              </S.BodyContent>
            </S.BodyWrapper>
          </Accordion>

          <Accordion icon={'Plus'} title={'Hawaiâ€™i'}>
            <S.BodyWrapper></S.BodyWrapper>
            <S.BodyWrapper>
              <S.BodyContent>
                <S.BodyText>
                  <b>Shifted Energy</b> will cover the costs of SPAN Panel in
                  full (plus up to $4,000 for installation) for customers who
                  agree to share SPAN Panel data and control.
                </S.BodyText>
                <S.LearnMoreButton>Learn more</S.LearnMoreButton>
              </S.BodyContent>
              <S.BodyContent>
                <S.BodyText>
                  <b>Hawaiian Electric</b> customers can save $500 on SPAN Panel
                  product costs by becoming a HECO Utility rate payer.
                </S.BodyText>
                <S.LearnMoreButton>Learn more</S.LearnMoreButton>
              </S.BodyContent>
            </S.BodyWrapper>
          </Accordion>
        </div>
        <div>
          <S.TextFooter>
            Any incentives, grants, or rebates provided by a government or
            public utility may affect the tax credits listed above.
          </S.TextFooter>
        </div>
      </S.Content>
    </S.Wrapper>
  )
}

export default LocalRebates
