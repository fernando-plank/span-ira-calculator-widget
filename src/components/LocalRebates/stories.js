import React from 'react';
import LocalRebates from '.';
export default {
    title: 'LocalRebates',
    component: LocalRebates
};
export const Default = () => React.createElement(LocalRebates, null);
