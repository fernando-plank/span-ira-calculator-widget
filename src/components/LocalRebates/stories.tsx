import React from 'react'
import { Story, Meta } from '@storybook/react'

import LocalRebates from '.'

export default {
  title: 'LocalRebates',
  component: LocalRebates
} as Meta

export const Default: Story = () => <LocalRebates />
