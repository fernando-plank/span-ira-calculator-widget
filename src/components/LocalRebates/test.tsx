import { screen } from '@testing-library/react'
import { render } from '@utils/test-utils'

import LocalRebates from '.'

jest.mock('../Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

describe('<LocalRebates />', () => {
  it('should render the title', () => {
    const { container } = render(<LocalRebates />)

    expect(screen.getByTestId('local-rebates-title')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
