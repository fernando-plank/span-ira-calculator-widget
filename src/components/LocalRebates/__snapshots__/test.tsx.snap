// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocalRebates /> should render the title 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: left;
  color: #1F1F1F;
  background: #EFEFEF;
  width: 100%;
}

.c1 {
  font-style: normal;
  font-weight: 700;
  font-size: 48px;
  line-height: 56px;
  margin-bottom: 81px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 40px;
}

.c9 {
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  width: 100%;
}

.c10 {
  border: 2px solid #1F1F1F;
  border-radius: 20px;
  width: 50%;
  height: 40px;
  margin-left: 50px;
  font-style: normal;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;
}

.c11 {
  font-style: normal;
  font-weight: 500;
  margin-top: 50px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

align-items:baseline .c2 > svg {
  cursor: pointer;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  opacity: 0;
  max-height: 0;
  pointer-events: none;
  -webkit-transition: max-height 0.2s ease-out;
  transition: max-height 0.2s ease-out;
  margin-bottom: 17px;
}

.c6.active {
  max-height: -webkit-min-content;
  max-height: -moz-min-content;
  max-height: min-content;
  opacity: 1;
  pointer-events: auto;
}

.c4 {
  font-style: normal;
  font-weight: 700;
  font-size: 32px;
  line-height: 40px;
  width: 80%;
  margin-bottom: 17px;
}

.c5 {
  height: 1px;
  background: #B0B0B1;
  border-radius: 2px;
  margin-bottom: 17px;
}

@media (min-width:320px) {
  .c0 {
    padding: 56px 24px;
  }
}

@media (min-width:1024px) {
  .c0 {
    padding: 96px 64px;
  }
}

@media (min-width:320px) {
  .c7 {
    width: 100%;
  }
}

@media (min-width:1024px) {
  .c7 {
    width: 50%;
  }
}

@media (min-width:320px) {
  .c8 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:1024px) {
  .c8 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:320px) {
  .c10 {
    -webkit-align-self: end;
    -ms-flex-item-align: end;
    align-self: end;
    margin-top: 20px;
  }
}

@media (min-width:1024px) {
  .c10 {
    -webkit-align-self: start;
    -ms-flex-item-align: start;
    align-self: start;
    margin-top: 0px;
  }
}

@media (min-width:320px) {
  .c11 {
    width: 100%;
    font-size: 16px;
    line-height: 24px;
  }
}

@media (min-width:1024px) {
  .c11 {
    width: 50%;
    font-size: 12px;
    line-height: 16px;
  }
}

@media (min-width:320px) {
  .c6 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:1024px) {
  .c6 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  class="c0"
>
  <div>
    <h1
      class="c1"
      data-testid="local-rebates-title"
    >
      Local utility electrification rebates
    </h1>
  </div>
  <div>
    <div
      class="c2 c3"
    >
      <h2
        class="c4"
        data-testid="accordion-title"
      >
        California
      </h2>
      <div>
        mocked icon
      </div>
    </div>
    <div
      class="c5"
    />
    <div
      class="c6"
      data-testid="accordion-body"
    >
      <div
        class="c7"
      />
      <div
        class="c7"
      >
        <div
          class="c8"
        >
          <p
            class="c9"
          >
            <b>
              Silicon Valley Power
            </b>
             customers can save up to $2,500 on SPAN as part of a larger home electrification project.
          </p>
          <button
            class="c10"
          >
            Learn more
          </button>
        </div>
        <div
          class="c8"
        >
          <p
            class="c9"
          >
            <b>
              Sacramento Municipal Utility District
            </b>
             (SMUD) customers can save up to $2,500 as part of a larger home electrification project.
          </p>
          <button
            class="c10"
          >
            Learn more
          </button>
        </div>
      </div>
    </div>
    <div
      class="c2 c3"
    >
      <h2
        class="c4"
        data-testid="accordion-title"
      >
        Hawai’i
      </h2>
      <div>
        mocked icon
      </div>
    </div>
    <div
      class="c5"
    />
    <div
      class="c6"
      data-testid="accordion-body"
    >
      <div
        class="c7"
      />
      <div
        class="c7"
      >
        <div
          class="c8"
        >
          <p
            class="c9"
          >
            <b>
              Shifted Energy
            </b>
             customers can have the costs of SPAN Panel covered, plus up to $4,000 for install when they agree to share SPAN Panel data and control with Shifted Energy.
          </p>
          <button
            class="c10"
          >
            Learn more
          </button>
        </div>
        <div
          class="c8"
        >
          <p
            class="c9"
          >
            <b>
              Hawaiian Electric
            </b>
             customers can save $500 on SPAN Panel product costs by becoming a HECO Utility rate payer.
          </p>
          <button
            class="c10"
          >
            Learn more
          </button>
        </div>
      </div>
    </div>
  </div>
  <div>
    <p
      class="c11"
    >
      We’re constantly updating this list, so check back to see if your utility provider provides localized incentives for getting SPAN.
    </p>
  </div>
</div>
`;
